runtime/src/main/scala/akka/grpc/internal/PlayRouter.scala

```diff
@@ -38,19 +32,10 @@ import scala.compat.java8.OptionConverters._
  *
  * INTERNAL API
  */
-@InternalApi abstract class PlayRouter(mat: Materializer, serviceName: String) extends play.api.routing.Router {
-
-  private val prefix = s"/$serviceName"
-
-  /**
-   * INTERNAL API
-   */
-  @InternalApi
-  protected def createHandler(serviceName: String, mat: Materializer): HttpRequest => Future[HttpResponse]
+@InternalApi abstract class PlayRouter(prefix: String, underlyingHandler: HttpRequest => Future[HttpResponse]) extends play.api.routing.Router {
```

Comment:
I removed a lot of the logic from this class and just pass things in as constructor arguments.